glib/src/regex.rs

```diff
@@ -0,0 +1,172 @@
+            let mut error = ptr::null_mut();
+            let is_ok = ffi::g_regex_match_all_full(
+                self.to_glib_none().0,
+                string.to_glib_none().0,
+                string_len,
+                start_position,
+                match_options.into_glib(),
+                &mut match_info,
+                &mut error,
+            );
+            debug_assert_eq!(is_ok == crate::ffi::GFALSE, !error.is_null());
+            if error.is_null() {
+                Ok(from_glib_full(match_info))
+            } else {
+                Err(from_glib_full(error))
+            }
+        }
+    }
+
+    #[doc(alias = "g_regex_match_full")]
+    pub fn match_full(
+        &self,
+        string: &str,
+        start_position: i32,
+        match_options: RegexMatchFlags,
+    ) -> Result<MatchInfo, crate::Error> {
+        let string_len = string.len() as _;
+        unsafe {
+            let mut match_info = ptr::null_mut();
+            let mut error = ptr::null_mut();
+            let is_ok = ffi::g_regex_match_full(
+                self.to_glib_none().0,
+                string.to_glib_none().0,
+                string_len,
+                start_position,
+                match_options.into_glib(),
+                &mut match_info,
+                &mut error,
+            );
+            debug_assert_eq!(is_ok == crate::ffi::GFALSE, !error.is_null());
+            if error.is_null() {
+                Ok(from_glib_full(match_info))
+            } else {
+                Err(from_glib_full(error))
+            }
+        }
+    }
+
+    #[doc(alias = "g_regex_replace_literal")]
+    pub fn replace_literal(
+        &self,
+        string: &str,
+        start_position: i32,
+        replacement: &str,
+        match_options: RegexMatchFlags,
+    ) -> Result<crate::GString, crate::Error> {
+        let string_len = string.len() as _;
+        unsafe {
+            let mut error = ptr::null_mut();
+            let ret = ffi::g_regex_replace_literal(
+                self.to_glib_none().0,
+                string.to_glib_none().0,
+                string_len,
+                start_position,
+                replacement.to_glib_none().0,
+                match_options.into_glib(),
+                &mut error,
+            );
+            if error.is_null() {
+                Ok(from_glib_full(ret))
+            } else {
+                Err(from_glib_full(error))
+            }
+        }
+    }
+
+    #[doc(alias = "g_regex_split_full")]
+    pub fn split_full(
+        &self,
+        string: &str,
+        start_position: i32,
+        match_options: RegexMatchFlags,
+        max_tokens: i32,
+    ) -> Result<Vec<crate::GString>, crate::Error> {
+        let string_len = string.len() as _;
+        unsafe {
+            let mut error = ptr::null_mut();
+            let ret = ffi::g_regex_split_full(
+                self.to_glib_none().0,
+                string.to_glib_none().0,
```

Comment:
You probably want to implement all these like

```rust
string.as_ptr() as *const _,
string.len() as _,
```

otherwise the `to_glib_none()` will always make a copy just to add the `NUL`-terminator, which is not even needed here because you explicitly provide the length anyway.